pyfolio/utils.py

```diff
@@ -574,6 +577,7 @@ def load_portfolio_risk_factors(filepath_prefix=None, start=None, end=None):
 }
 
 
+@deprecated(msg=DATAREADER_DEPRECATION_WARNING)
```

Comment:
Do we want to deprecate this functionality? This seems potentially like a nice way to allow a user that has their own reliable source of returns data (including, for example Quantopian), to register their preferred method of returns acquisition without having to explicitly pass returns to every pyfolio function.

I'm not sure how much of a burden that actually is in practice, but we seemed to previously think it was significant enough that we wanted to fetch returns implicitly.

If we decide to leave this, then we probably would want to go back to calling `get_symbol_rets` (maybe renamed to something like `get_default_symbol_rets`) in the places where we used to do so previously, but we'd have the default be to just return None, and we'd have to be prepared to handle a return of None.

If we remove this, we need to make sure that we update anywhere internal on Quantopian where we depend on having registered a function here. In particular, we probably need to make sure we're passing benchmark_rets everywhere that it's desired now.

Reply:
@ssanderson Thinking about it, I agree that this functionality might still be useful, and we shouldn't deprecate it. However, I'm not sure what you mean by "go back to calling `get_symbol_rets`". We currently do not call this function at all in `pyfolio`.

Reply:
@eigenfoo on pyfolio master I see at least 5 call-sites of this in the library, plus several in the example notebooks:

```
git --no-pager grep -n -e get_symbol_rets -- 
Binary file pyfolio/examples/bayesian.ipynb matches
Binary file pyfolio/examples/fama_french_benchmark.ipynb matches
Binary file pyfolio/examples/pyfolio_talk_slides.ipynb matches
Binary file pyfolio/examples/single_stock_example.ipynb matches
pyfolio/tears.py:193:        benchmark_rets = utils.get_symbol_rets('SPY')
pyfolio/tears.py:342:        benchmark_rets = utils.get_symbol_rets('SPY')
pyfolio/tears.py:501:        benchmark_rets = utils.get_symbol_rets('SPY')
pyfolio/tears.py:956:        benchmark_rets = utils.get_symbol_rets('SPY')
pyfolio/tears.py:1149:            utils.get_symbol_rets('SPY',
```

Reply:
@ssanderson all 5 of those calls have been removed in this PR.

The example notebooks is not something I've thought of, though. It's out of scope for this PR, but we'll need to revisit it later.